db/db_iter.h

```diff
@@ -313,13 +313,17 @@ class DBIter final : public Iterator {
 
   bool SetValueAndColumnsFromEntity(Slice slice);
 
+  bool SetValueAndColumnsFromMergeResult(const Status& merge_status,
+                                         ValueType result_type);
+
   void ResetValueAndColumns() {
     value_.clear();
     wide_columns_.clear();
   }
 
   // If user-defined timestamp is enabled, `user_key` includes timestamp.
-  bool Merge(const Slice* val, const Slice& user_key);
+  bool MergeNone(const Slice& user_key);
```

Comment:
It's a bit hard to guess what's a valid use case of this API without further reading the code. Especially since this is no different from `MergePlain(nullptr, key)`.

Reply:
Good point; let me add some comments (and maybe adjust the names) to explain the usage of these. BTW, the main rationale of this change is to clearly differentiate between the plain base value and no base value cases; because of this, you can't actually call `MergePlain` with `nullptr` (since it's a reference now, not a pointer).